internal/services/postgres/postgresql_flexible_server_firewall_rules_resource.go

```diff
@@ -0,0 +1,464 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package postgres
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/postgresql/2022-12-01/firewallrules"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/postgresql/2023-06-01-preview/servers"
+	"github.com/hashicorp/go-azure-sdk/sdk/client/pollers"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type FirewallRuleWithId struct {
+	Id           firewallrules.FirewallRuleId
+	FirewallRule firewallrules.FirewallRule
```

Comment:
These are missing struct tags which is what's causing the error you're seeing, but it also looks like this isn't in use? So this could be deleted
```suggestion
	Id           firewallrules.FirewallRuleId `tfschema:"id"`
	FirewallRule firewallrules.FirewallRule `tfschema:"firewall_rule"`
```